Lib/test/test_ctypes/_support.py

```diff
@@ -22,3 +18,115 @@ class _X(Structure):
+                if not is_struct:
+                    self.assertEqual(field.byte_offset, 0)
+
+                # byte_size
+                self.assertEqual(field.byte_size, ctypes.sizeof(field.type))
+                self.assertGreaterEqual(field.byte_size, 0)
+                self.assertLessEqual(field.byte_offset + field.byte_size,
+                                     cls_size)
+
+                # size
+                self.assertGreaterEqual(field.size, 0)
+                if is_bitfield:
+                    # size has backwards-compatible bit-packed info
+                    if hasattr(cls, '_swappedbytes_'):
+                        offset_for_size = (8 * field.byte_size
+                                           - field.bit_offset
+                                           - field.bit_size)
+                    else:
+                        offset_for_size = field.bit_offset
+                    expected_size = (field.bit_size << 16) + offset_for_size
+                    self.assertEqual(field.size, expected_size)
+                else:
+                    # size == byte_size
+                    self.assertEqual(field.size, field.byte_size)
+
+                # is_bitfield (bool)
+                self.assertIs(field.is_bitfield, is_bitfield)
+
+                # bit_offset
+                if is_bitfield:
+                    self.assertGreaterEqual(field.bit_offset, 0)
+                    self.assertLessEqual(field.bit_offset + field.bit_size,
+                                         field.byte_size * 8)
+                else:
+                    self.assertEqual(field.bit_offset, 0)
+                if not is_struct:
+                    self.assertEqual(field.bit_offset, 0)
+
+                # bit_size
+                if is_bitfield:
+                    self.assertGreaterEqual(field.bit_size, 0)
+                    self.assertLessEqual(field.bit_size, field.byte_size * 8)
+                    [requested_bit_size] = rest_of_tuple
+                    self.assertEqual(field.bit_size, requested_bit_size)
+                else:
+                    self.assertEqual(field.bit_size, field.byte_size * 8)
+
+                # is_anonymous (bool)
+                self.assertIs(field.is_anonymous, name in anon_names)
+
+                # field is not overlapping earlier members in a struct.
+                # (this assumes fields are laid out in order)
+                self.assertGreaterEqual(
+                    field.byte_offset * 8 + field.bit_offset,
+                    next_bit)
+                next_bit = (field.byte_offset * 8
+                            + field.bit_offset
+                            + field.bit_size)
+                # field is inside cls
+                self.assertLessEqual(next_bit, cls_size * 8)
+
+                if not is_struct:
+                    # union fields may overlap
+                    next_bit = 0
+            except Exception as e:
```

Comment:
Off-topic but this makes me wonder whether a re-raising context would be interesting:

```py
with add_note_on_failure(f'while checking field {name!r}: {field}'):
	...
```

I don't know if we're adding notes a lot in the test folder though so maybe not that useful. OTOH, this could be a `subTest` instead.